### PR TITLE
Deprecate Brave Ads DoesAdTypeSupportFrequencyCapping

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/20983

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Confirm conversion frequency cap is working as expected
- Confirm daily frequency cap (campaign) is working as expected
- Confirm per day frequency cap (creative set) is working as expected
- Confirm per hour frequency cap (creative_instance_id) is working as expected
- Confirm per month frequency cap (creative set) is working as expected
- Confirm per week frequency cap (creative set) is working as expected
- Confirm total max frequency cap (creative set) is working as expected
- Confirm transferred frequency cap (campaign) is working as expected

^^^ for ad notifications, new tab page ads and inline content ads (also test different ad types within the same campaign/creative which should have their own caps.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ads/inline_content_ads/inline_content_ad.cc [hunk 1]
```diff
@@ -73,25 +73,31 @@ void InlineContentAd::FireEvent(
...
     const mojom::InlineContentAdEventType event_type) {
   database::table::AdEvents database_table;
-  database_table.GetAll([=](const bool success, const AdEventList& ad_events) {
-    if (!success) {
-      BLOG(1, "Inline content ad: Failed to get ad events");
-      NotifyInlineContentAdEventFailed(uuid, creative_instance_id, event_type);
-      return;
-    }
+  database_table.GetForType(
+      mojom::AdType::kInlineContentAd,
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/inline_content_ads/inline_content_ad.cc:77]: We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps

File: vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/inline_content_ads/eligible_inline_content_ads_v1.cc [hunk 2]
```diff
@@ -40,21 +40,24 @@ void EligibleAdsV1::GetForUserModel(
...
-    const int max_count = features::GetBrowsingHistoryMaxCount();
-    const int days_ago = features::GetBrowsingHistoryDaysAgo();
-    AdsClientHelper::Get()->GetBrowsingHistory(
-        max_count, days_ago, [=](const BrowsingHistoryList& browsing_history) {
-          GetEligibleAds(user_model, dimensions, ad_events, browsing_history,
-                         callback);
-        });
-  });
+  database_table.GetForType(
+      mojom::AdType::kInlineContentAd,
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/inline_content_ads/eligible_inline_content_ads_v1.cc:44]: We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps

File: vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/inline_content_ads/eligible_inline_content_ads_v2.cc [hunk 3]
```diff
@@ -39,21 +39,24 @@ void EligibleAdsV2::GetForUserModel(
...
-    const int max_count = features::GetBrowsingHistoryMaxCount();
-    const int days_ago = features::GetBrowsingHistoryDaysAgo();
-    AdsClientHelper::Get()->GetBrowsingHistory(
-        max_count, days_ago, [=](const BrowsingHistoryList& browsing_history) {
-          GetEligibleAds(user_model, ad_events, browsing_history, dimensions,
-                         callback);
-        });
-  });
+  database_table.GetForType(
+      mojom::AdType::kInlineContentAd,
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/inline_content_ads/eligible_inline_content_ads_v2.cc:43]: We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps

File: vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/new_tab_page_ads/eligible_new_tab_page_ads_v1.cc [hunk 4]
```diff
@@ -39,20 +39,23 @@ void EligibleAdsV1::GetForUserModel(
...
-
-    const int max_count = features::GetBrowsingHistoryMaxCount();
-    const int days_ago = features::GetBrowsingHistoryDaysAgo();
-    AdsClientHelper::Get()->GetBrowsingHistory(
-        max_count, days_ago, [=](const BrowsingHistoryList& browsing_history) {
-          GetEligibleAds(user_model, ad_events, browsing_history, callback);
-        });
-  });
+  database_table.GetForType(
+      mojom::AdType::kNewTabPageAd,
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/new_tab_page_ads/eligible_new_tab_page_ads_v1.cc:43]: We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps

File: vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/new_tab_page_ads/eligible_new_tab_page_ads_v2.cc [hunk 5]
```diff
@@ -38,20 +38,23 @@ void EligibleAdsV2::GetForUserModel(
...
-
-    const int max_count = features::GetBrowsingHistoryMaxCount();
-    const int days_ago = features::GetBrowsingHistoryDaysAgo();
-    AdsClientHelper::Get()->GetBrowsingHistory(
-        max_count, days_ago, [=](const BrowsingHistoryList& browsing_history) {
-          GetEligibleAds(user_model, ad_events, browsing_history, callback);
-        });
-  });
+  database_table.GetForType(
+      mojom::AdType::kNewTabPageAd,
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/new_tab_page_ads/eligible_new_tab_page_ads_v2.cc:42]: We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/total_max_exclusion_rule_unittest.cc [hunk 6]
```diff
@@ -67,35 +67,6 @@ TEST_F(BatAdsTotalMaxExclusionRuleTest, AllowAdIfDoesNotExceedCap) {
...
-  ad_events.push_back(ad_event_3);
-
-  // Act
-  TotalMaxExclusionRule frequency_cap(ad_events);
-  const bool should_exclude = frequency_cap.ShouldExclude(creative_ad);
-
-  // Assert
-  EXPECT_FALSE(should_exclude);
-}
-
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/total_max_exclusion_rule_unittest.cc:98]: We can ignore this use case as there will be a change where campaigns will only have a single type

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/exclusion_rule_util.h [hunk 7]
```diff
@@ -14,10 +14,6 @@
 
 namespace ads {
 
-class AdType;
-
-bool DoesAdTypeSupportFrequencyCapping(const AdType& type);
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/exclusion_rule_util.h:19]: Filtering by ad type is no longer required as with this change we only pass through the ad types to calculate the eligible ads

**Review by aseren — APPROVED** [review]

**Pull request merged**